src/main/scala/db/migration/V31__ConvertBrightcoveIds.scala

```diff
@@ -0,0 +1,206 @@
+/*
+ * Part of NDLA draft-api.
+ * Copyright (C) 2021 NDLA
+ *
+ * See LICENSE
+ */
+
+package db.migration
+
+import no.ndla.draftapi.DraftApiProperties.prop
+import no.ndla.network.model.HttpRequestException
+import no.ndla.validation.{ValidationException, ValidationMessage}
+import org.flywaydb.core.api.migration.{BaseJavaMigration, Context}
+import org.json4s
+import org.json4s.{DefaultFormats, Formats}
+import org.json4s.JsonAST.{JArray, JString}
+import org.json4s.native.JsonMethods.{compact, parse, render}
+import org.json4s.native.Serialization.read
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Entities.EscapeMode
+import org.postgresql.util.PGobject
+import scalaj.http.{Http, HttpRequest, HttpResponse}
+import scalikejdbc.{DB, DBSession, _}
+import org.joda.time.DateTime
+
+import scala.util.{Failure, Success, Try}
+
+case class BrightcoveToken(access_token: String, expires_in: Long)
+case class StoredToken(accessToken: String, expiresAt: Long)
+case class BrightcoveData(id: String)
+
+object BrightcoveApiClient {
+  implicit val jsonFormats: Formats = DefaultFormats
+
+  private lazy val BrightcoveAccountId = prop("NDLA_BRIGHTCOVE_ACCOUNT_ID")
+  private lazy val ClientId = prop("BRIGHTCOVE_API_CLIENT_ID")
+  private lazy val ClientSecret = prop("BRIGHTCOVE_API_CLIENT_SECRET")
+  private val timeout = 20 * 1000 // 20 Seconds
+
+  private var accessToken: Option[StoredToken] = None
+
+  def fetchBrightcoveVideo(id: String): Try[BrightcoveData] = {
+    refreshTokenIfInvalid().flatMap(token => fetch(id, token))
+  }
+
+  private def refreshToken(): Try[StoredToken] = {
+    fetchAccessToken().map(token => {
+      val temp = StoredToken(
+        token.access_token,
+        token.expires_in + (new DateTime().getMillis / 1000)
+      )
+      accessToken = Some(temp)
+      temp
+    })
+  }
+
+  private def refreshTokenIfInvalid(): Try[StoredToken] = {
+    accessToken match {
+      case Some(storedToken) if ((new DateTime().getMillis / 1000) < storedToken.expiresAt) => Success(storedToken)
```

Comment:
Jeg tror kanskje det er smart å ha en liten sikkerhetsmargin her sånn at vi ikke sjekker _akkurat_ rett før tokenet går ut også skjer requesten etter at tokenet har gått ut. Så kanskje noe sånt?
```suggestion
      case Some(storedToken) if (new DateTime().getMillis / 1000) < storedToken.expiresAt - 10 => Success(storedToken)
```